app/jobs/get_next_ai_message_job.rb

```diff
@@ -199,6 +209,12 @@ def call_tools_before_wrapping_up
       index: index += 1
     )
 
+    unless url_of_dalle_generated_image.nil?
+      d = Document.new
+      d.file.attach(io: URI.open(url_of_dalle_generated_image), filename: 'image.png')
+      assistant_reply.documents << d
+    end
```

Comment:
We should assume that `@message.content_tool_calls` can have multiple Dalle toolbox calls. This means that within `msgs.each do |tool_message| ` each of those calls will set the `url_of_dalle_generated_image` overwriting the previous.

So let's collect multiple `url_of_dalle_generated_image` into an array so that in this block here we push multiple documents into the assistant_reply.